homeassistant/components/switcher_kis/switch.py

```diff
@@ -0,0 +1,294 @@
+from homeassistant.const import (CONF_FRIENDLY_NAME, CONF_ICON, CONF_NAME,
+                                 STATE_OFF, STATE_ON, STATE_UNKNOWN)
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import PlatformNotReady
+from homeassistant.helpers.entity import async_generate_entity_id
+
+from . import ENTITY_ID_FORMAT as SWITCHER_KIS_FORMAT
+from ._service_registration import (DISCOVERY_CONFIG, DISCOVERY_DEVICE,
+                                    async_register_switch_entity)
+
+DEPENDENCIES = ['switcher_kis']
+
+_LOGGER = getLogger(__name__)
+
+ENTITY_ID_FORMAT = SWITCH_FORMAT.format(SWITCHER_KIS_FORMAT)
+
+ATTR_IP_ADDRESS = 'ip_address'
+ATTR_ELECTRIC_CURRNET = 'electric_current'
+ATTR_REMAINING_TIME = 'remaining_time'
+ATTR_AUTO_OFF_SET = 'auto_off_set'
+ATTR_LAST_DATA_UPDATE = 'last_data_update'
+ATTR_LAST_STATE_CHANGE = 'last_state_change'
+ATTR_DEVICE_NAME = 'device_name'
+
+PROPERTIES_TO_ATTRIBUTES = {
+    'current_power_w': ATTR_CURRENT_POWER_W,
+    'device_ip_addr': ATTR_IP_ADDRESS,
+    'electric_current': ATTR_ELECTRIC_CURRNET,
+    'remaining_time': ATTR_REMAINING_TIME,
+    'auto_off_set': ATTR_AUTO_OFF_SET,
+    'last_data_update': ATTR_LAST_DATA_UPDATE,
+    'last_state_change': ATTR_LAST_STATE_CHANGE,
+    'device_name': ATTR_DEVICE_NAME
+}
+
+
+async def async_setup_platform(hass: HomeAssistant, config: Dict,
+                               async_add_entities: Callable,
+                               discovery_info: Dict) -> None:
+    """Set up the switcher platform for the switch component."""
+    if discovery_info.get(DISCOVERY_CONFIG):
+        name = str(discovery_info[DISCOVERY_CONFIG].get(CONF_NAME))
+
+        friendly_name = discovery_info[DISCOVERY_CONFIG].get(
+            CONF_FRIENDLY_NAME, name.title())
+
+        icon = discovery_info[DISCOVERY_CONFIG].get(CONF_ICON)
+
+    else:
+        raise PlatformNotReady("No config data found")
+
+    if discovery_info.get(DISCOVERY_DEVICE):
+        switcher_entity = SwitcherControl(hass, name, friendly_name, icon,
+                                          discovery_info.get(DISCOVERY_DEVICE))
+
+    else:
+        raise PlatformNotReady("No device data discoverd")
+
+    async_add_entities([switcher_entity], False)
+
+    try:
+        await wait_for(async_register_switch_entity(switcher_entity),
+                       timeout=1)
+
+    except AsyncioTimeoutError:
+        raise PlatformNotReady("Unable to register for data updates")
+
+    return None
+
+
+class SwitcherControl(SwitchDevice):
+    """Home Assistant entity.
+
+    Data updates is handled by the bridge thread,
+    Therefore no polling by ha is needed.
+    """
+
+    def __init__(self, hass: HomeAssistant, name: str, friendly_name: str,
+                 icon: Optional[str], device_data: SwitcherV2Device) -> None:
+        """Initialize the entity."""
+        self._hass = hass
+        self.entity_id = async_generate_entity_id(
```

Comment:
Do not overwrite this property.  `entity_id` is generated automatically based on the platform domain and name.